samples/excel/90-scenarios/currency-converter.yaml

```diff
@@ -0,0 +1,178 @@
+              let priceInCcy = row.values[0][tblSetup['Price (CCY)']];
+              let trxDate = row.values[0][tblSetup['Trx Date']]
+              let ccy = row.values[0][tblSetup['CCY']];
+              let baseCcy = row.values[0][tblSetup['Base']]
+              if (priceInBase === "") {
+                let dateMoment = moment.fromOADate(trxDate);
+                let period = dateMoment.format("YYYY-MM-DD");
+                let queryUrl = convertValue(ccy, baseCcy, period, period);
+                let response = await fetch(queryUrl);
+                if (response.ok) {
+                  let result = await response.json();
+                  let converted = priceInCcy * result.rates[baseCcy];
+                  priceBaseRange.values = [[converted]];
+                  context.sync();
+                }
+              } else {
+                console.info('Row: ' + idx + ' has been converted. Skipped.')
+              }
+          });
+          
+        });
+      }
+
+      /** Check if Table is Valid, ie has all required column **/
+      function IsTableValid(items: Excel.TableColumn[]) {
+        // Build the column index, search table header
+        // CCY - origin currency
+        // Price (CCY) - price in origin currency
+        // Trx Date - transaction date
+        // Base  - home currency
+        // Price in Base - price in base currency
+        items.forEach((col, idx, arr) => {
+          tblSetup[col.name] = idx;
+        });
+        if (tblSetup['CCY'] === undefined || tblSetup['Price (CCY)'] === undefined || tblSetup['Trx Date'] === undefined || tblSetup['Base'] === undefined || tblSetup['Price in Base'] === undefined) {
+          return false;
+        }
+        
+        return true;
+      }
+
+      /** Request currency exchange on specific date **/      
+      function convertValue(ccyOrig, ccyBase, start, end) {
+        // GET https://api.exchangeratesapi.io/latest?symbols=SGD,USD&base=SGD&start_at=2019-10-05&end_at=2019-10-05
+        const baseUrl = "https://api.exchangeratesapi.io/latest";
+        let query = "?base=" + ccyOrig + "&symbols=" + ccyBase + "," + ccyOrig + "&start_at=" + start + "&end_at=" + end;
+        return baseUrl + query;
+      }
+
+      /** Default helper for invoking an action and handling errors. */
+      async function tryCatch(callback) {
+        try {
+          await callback();
+        } catch (error) {
+          // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+          console.error(error);
+        }
+      }
+      /* Create sample data */
+      /* Default header */
+      /* "CCY","Price (CCY)","Trx Date","Base","Price in Base" */
+      async function setupSample() {
+        await Excel.run(async (context) => {
+          context.workbook.worksheets.getItemOrNullObject("Sample").delete();
+          const sheet = context.workbook.worksheets.add("Sample");
+          let salesTable = sheet.tables.add("A1:F1", true);
```

Comment:
The table is less about sales and more of a transaction history. Maybe call it `transactionTable`

Reply:
Updated.